compiler/function-colors.cpp

```diff
@@ -0,0 +1,84 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2021 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "function-colors.h"
+
+using namespace function_palette;
+
+colors_t ColorContainer::colors() {
+  return data;
+}
+
+const std::vector<color_t> &ColorContainer::sep_colors() {
+  return sep_colors_;
+}
+
+void ColorContainer::add(color_t color) {
+  data |= color;
+  count++;
+  sep_colors_.push_back(color);
+}
+
+size_t ColorContainer::size() const noexcept {
```

Comment:
It's better to implement such very small functions right in the .h file, so that they can be inlined